### ISSUE TITLE
Network Seed UI suggestion

### ISSUE THREAD

**Issue by TechSupport26** [holochain/launcher-tauri#200]
**Is your feature request related to a problem? Please describe.**
When you go to copy a Network Seed a popup appears saying "Copied!" This is great, however if the Network Seed is blank then nothing happens. (I think that is good however it is not so great for knowing that nothing was copied)

**Describe the solution you'd like**
I would suggest a popup that says something. E.g "There was nothing to copy"

**Describe alternatives you've considered**
N/A
**Additional context**
Let me know if there are any questions.

**Comment by matthme** [comment]
Thanks! Will try to add this to the next release.

**Comment by matthme** [comment]
Hmm...it is actually copying the empty network seed, it's just not showing a tooltip. @TechSupport26 what do you think is better behavior, a tooltip saying that there's nothing to copy and the clipboard remains untouched or actually copying an empty string and displaying the "Copied" tooltip?

**Comment by TechSupport26** [comment]
TL;DR at the bottom.

> Hmm...it is actually copying the empty network seed, it's just not showing a tooltip.

From a users perspective I am not able to see that anything is copied or stored in the clipboard (as there is nothing to copy)


> a tooltip saying that there's nothing to copy and the clipboard remains untouched or actually copying an empty string and displaying the "Copied" tooltip?

My initial thought is for the clipboard to remain untouched.

(I believe that this next part can be safely ignored however I will leave it in for "possible consideration".)
However I have also thought about the security of this as, if someone were to be screen recording then having the "Copied" tooltip would be needed as to not give away the (supposedly empty) Network Seed. In saying that if it were to be empty then (if they intended to share the Network Seed) it would be fairly obvious that it was not intended for private use and anyone can access it and it and so, poses no security threat that I can think of. (I think I am getting the idea across.)

TL;DR
My opinion on what would be better behaviour is to not copy anything (as there is nothing to copy) and display a tooltip to the user saying "there's nothing to copy" rather than copying an empty string and displaying the "Copied" tooltip as this could lead to a bit of confusion when a user thought they copied something when in reality nothing happened.

I hope this helps and that I did not over complicate this.

**Comment by matthme** [comment]
Good points. I haven't thought of the screen sharing example actually but I agree that an empty network seed should probably be considered public anyway. Went with your proposed solution now.

**Comment by TechSupport26** [comment]
> an empty network seed should probably be considered public anyway.

Hehe. You worded and condensed that part of what I was saying really well.

Will this automatically close when the next release is made or should one of us close it manually after the release?

**Comment by matthme** [comment]
I think we'll have to close it manually.

**Comment by matthme** [comment]
Released both for [holochain 0.1](https://github.com/holochain/launcher/releases/tag/v0.10.2) and [holochain 0.2](https://github.com/holochain/launcher/releases/tag/v0.11.1) now.
